src/components/vouchers/VoucherRedeem.tsx

```diff
@@ -0,0 +1,298 @@
+import React, { useEffect, useState } from 'react';
+import { Modal, StyleSheet, TouchableOpacity, View } from 'react-native';
+import CircularProgress from 'react-native-circular-progress-indicator';
+
+import { Icon, colors, normalize, texts } from '../../config';
+import { Button } from '../Button';
+import { BoldText, RegularText } from '../Text';
+import { Touchable } from '../Touchable';
+import { Wrapper, WrapperRow } from '../Wrapper';
+import { Checkbox } from '../Checkbox';
+import { TQuota } from '../../types';
+
+const defaultTime = 15 * 60; // 15 * 60 sec.
+
+export const VoucherRedeem = ({ quota }: { quota: TQuota }) => {
+  const [isVisible, setIsVisible] = useState(false);
+  const [remainingTime, setRemainingTime] = useState(defaultTime);
+  const [isRedeemVoucher, setIsRedeemVoucher] = useState(false);
+  const [isChecked, setIsChecked] = useState(false);
+  const [quantity, setQuantity] = useState(1);
+
+  // TODO: set voucher availability
+  const [isExpiredVoucher, setIsExpiredVoucher] = useState(false);
+
+  const { maxPerPerson } = quota;
+
+  useEffect(() => {
+    if (isRedeemVoucher) {
+      const interval = setInterval(() => {
+        if (remainingTime > 0) {
+          setRemainingTime(remainingTime - 1);
+        } else {
+          clearInterval(interval);
+          setIsExpiredVoucher(true);
+        }
+      }, 1000);
+
+      return () => clearInterval(interval);
+    } else {
+      setRemainingTime(defaultTime);
+    }
+  }, [remainingTime, isRedeemVoucher]);
+
+  const minutes = Math.floor(remainingTime / 60);
+  const seconds = remainingTime % 60;
+
+  return (
+    <>
+      {/* TODO: button availability will be adjusted according to voucher availability */}
+      <Button title={texts.voucher.detailScreen.redeem} onPress={() => setIsVisible(true)} />
+
+      <Modal animationType="slide" transparent visible={isVisible}>
```

Comment:
i suggest using the same as in DateTimePicker and WasteReminderSettings:

```
<Modal
  animationType="none"
  transparent
  visible={isModalVisible}
  supportedOrientations={['landscape', 'portrait']}
>
```

that transparent white background doesn't look good on sliding.
and sliding could suggest removal of the modal by sliding down. which we do not have.

what do you say?

Reply:
I can add this feature. But I need to change the `presentation` format. The slide down closing feature is only valid in `pageSheet` or `formSheet`.

I can make the background transparent. But as I see in Figma we have a transparent white background. That's why I made this addition.

Reply:
we do not want the feature to dismiss the modal and we do not want a transparent background.
so as i suggested the less effort is to make it not animated.

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/989/commits/cd8dff426b963376df5719fb1a20b9b6fa65fa8a